app/common/state/shieldState.js

```diff
@@ -0,0 +1,20 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const frameStateUtil = require('../../../js/state/frameStateUtil')
+const urlParse = require('../urlParse')
+
+function braveShieldsEnabled (frame) {
+  const lastCommittedURL = frameStateUtil.getLastCommittedURL(frame)
+  if (!lastCommittedURL) {
+    return false
+  }
+
+  const parsedUrl = urlParse(lastCommittedURL)
+  return !(parsedUrl.protocol !== 'https:' && parsedUrl.protocol !== 'http:')
```

Comment:
merging now, but just a nit that this would be easier to read in the positive form `parsedUrl === 'https'` or better yet use `urlutil.isHttpOrHttps`